click/testing.py

```diff
@@ -285,21 +286,24 @@ def invoke(self, cli, args=None, input=None, env=None,
                 cli.main(args=args or (),
                          prog_name=self.get_default_prog_name(cli), **extra)
             except SystemExit as e:
```

Comment:
Hmm, if we instead write `exit_code = e.code; if exit_code is None: exit_code = 0` and replace `e.code` with `exit_code` further down, we can spare the double-check for `None`